src/shogun/mathematics/linalg/LinalgBackendViennaCL.h

```diff
@@ -364,16 +364,40 @@ namespace shogun
 		/** ViennaCL matrix in-place elementwise product method */
 		template <typename T>
 		void element_prod_impl(
-		    const SGMatrix<T>& a, const SGMatrix<T>& b, SGMatrix<T>& result) const
+		    const SGMatrix<T>& a, const SGMatrix<T>& b, SGMatrix<T>& result,
+		    bool transpose_A, bool transpose_B) const
 		{
 			GPUMemoryViennaCL<T>* a_gpu = cast_to_viennacl(a);
 			GPUMemoryViennaCL<T>* b_gpu = cast_to_viennacl(b);
 			GPUMemoryViennaCL<T>* result_gpu = cast_to_viennacl(result);
 
-			result_gpu->data_matrix(a.num_rows, a.num_cols) =
-			    viennacl::linalg::element_prod(
-			        a_gpu->data_matrix(a.num_rows, a.num_cols),
-			        b_gpu->data_matrix(a.num_rows, a.num_cols));
+			if (transpose_A && transpose_B)
+				result_gpu->data_matrix(result.num_rows, result.num_cols) =
+				    viennacl::linalg::element_prod(
+				        viennacl::trans(
```

Comment:
cant we store the "trans" in a varialbe and then use that in the if-then-else. Then we don't copy the whole code block 3 times ...